### PR TITLE
BC-1767 Improve feature toggle flow for student visibility

### PR DESCRIPTION
**Pull request by davwas**
# Short Description
Changed student visibility switch state on new admin page based on new env var.

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-1767
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4239

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by blazejpass — APPROVED** [review]

File: src/components/organisms/administration/PrivacySettings.unit.js [hunk 1]
```diff
@@ -56,7 +56,6 @@ describe("PrivacySettings", () => {
 			envConfigModule.setEnvs({
 				FEATURE_ADMIN_TOGGLE_STUDENT_LERNSTORE_VIEW_ENABLED: false,
 				FEATURE_LERNSTORE_ENABLED: true,
-				TEACHER_STUDENT_VISIBILITY__IS_CONFIGURABLE: false,
```
Comment (dyedwiper) [inline src/components/organisms/administration/PrivacySettings.unit.js:59]: I think we need to add a test that checks that the switch is disabled when the env var is false.

Comment (dyedwiper) [reply src/components/organisms/administration/PrivacySettings.unit.js:59]: ...and also one that then checks the state of the switch depending on the env var for the default.

Comment (davwas) [reply src/components/organisms/administration/PrivacySettings.unit.js:59]: test added

File: src/components/organisms/administration/PrivacySettings.unit.js [hunk 2]
```diff
@@ -103,71 +137,174 @@ describe("PrivacySettings", () => {
...
+				});
+				const wrapper = mount(PrivacySettings, {
+					...createComponentMocks({
+						i18n: true,
+						vuetify: true,
+					}),
+					propsData: Object.assign(generateProps(), {
+						permissions: {
+							teacher: {
+								STUDENT_LIST: false,
```
Comment (dyedwiper) [inline src/components/organisms/administration/PrivacySettings.unit.js:266]: You don't need to set STUDENT_LIST for this test, do you? Also in the one below.

File: src/components/organisms/administration/PrivacySettings.unit.js [hunk 3]
```diff
@@ -35,11 +35,48 @@ describe("PrivacySettings", () => {
 	});
 
 	describe("env config", () => {
+		it("should enable student visibility switch", () => {
```
Comment (dyedwiper) [inline src/components/organisms/administration/PrivacySettings.unit.js:38]: I think the description should give a little more context, maybe like so:
```suggestion
		it("should enable student visibility switch according to env var", () => {
```

Comment (dyedwiper) [reply src/components/organisms/administration/PrivacySettings.unit.js:38]: ...also in the other tests.

File: src/components/organisms/administration/PrivacySettings.unit.js (hunk 2 above)
Comment (davwas) [reply src/components/organisms/administration/PrivacySettings.unit.js]: correct, this was set on the old tests and I forgot to clean it up, fixed now

File: src/components/organisms/administration/PrivacySettings.unit.js (hunk 3 above)
Comment (davwas) [reply src/components/organisms/administration/PrivacySettings.unit.js:38]: those tests are already described as 'env-config' and I wanted to keep the naming 'convention' that was already in the file, let me know if you still want me to rename them

File: src/locales/de.json [hunk 4]
```diff
@@ -670,6 +670,7 @@
 	"pages.administration.school.index.privacySettings.longText.lernStore": "Wenn diese Option nicht aktiviert ist, können die Schüler:innen nicht auf den Lern-Store zugreifen.",
 	"pages.administration.school.index.privacySettings.longText.chatFunction": "Sind Chats an deiner Schule aktiviert, können Team-Admins im jeweiligen Team sowie Lehrkräfte in ihren Kursen die Chatfunktion gezielt freischalten.",
 	"pages.administration.school.index.privacySettings.longText.videoConference": "Sind Videokonferenzen an deiner Schule aktiviert, können Lehrkräfte ihrem Kurs im Bereich Tools das Videokonferenz-Tool hinzufügen und dann von dort aus Videokonferenzen für alle Kursteilnehmer:innen starten. Team-Admins können die Videokonferenzfunktion im jeweiligen Team aktivieren. Team-Leiter:innen und Team-Admins können dann Videokonferenzen zu Terminen hinzufügen und starten.",
+	"pages.administration.school.index.privacySettings.longText.configurabilityInfoText": "Diese Option wird für die gesamte Instanz festgelegt und ist nicht konfigurierbar. Sie steuert, ob Lehrer eine Liste aller Schüler in Ihrer Schule sehen können.",
```
Comment (dyedwiper) [inline src/locales/de.json:673]:
```suggestion
	"pages.administration.school.index.privacySettings.longText.configurabilityInfoText": "Diese Option wird für die gesamte Instanz festgelegt und ist nicht konfigurierbar. Sie steuert, ob Lehrer eine Liste aller Schüler in deiner Schule sehen können.",
```

File: src/components/organisms/administration/PrivacySettings.unit.js (hunk 3 above)
Comment (dyedwiper) [reply src/components/organisms/administration/PrivacySettings.unit.js:38]: I don't like that it reads just "env-config should enable student visibility switch" or "env-config should disable student visibility switch". There needs be a condition. Maybe better than "according to env var" would be "if env var for configurability is true"...

File: src/locales/de.json (hunk 4 above)
Comment (davwas) [reply src/locales/de.json]: Fixed.

File: src/components/organisms/administration/PrivacySettings.unit.js (hunk 3 above)
Comment (davwas) [reply src/components/organisms/administration/PrivacySettings.unit.js]: I've restructured the tests in the file, please have a look again.

**Review by dyedwiper — APPROVED** [review]

**Review by blazejpass — APPROVED** [review]

**Pull request merged**